### PR TITLE
[HEAP-8830] Don't allow errors thrown by the Heap library to crash the client app.

### PR DESCRIPTION
**Pull request by timothyandrew**
- Errors are swallowed in production and surfaced in debug builds.
- This covers the public API and the autotracking code.

### CODE REVIEW

**Comment by timothyandrew** [comment]
Good point, I'll look into this. There are a few other things pending as well before this is ready for a full review:

- Allow crashes in debug builds, either by re-throwing the error or displaying a warning.
- Potentially disable tracking (with backoff) on crash.

**Comment by timothyandrew** [comment]
This should be ready for review. I added a second commit with a test covering some of the crash handling behavior, but this requires adding a method to the public `Heap` API, and I'm not sure if that's worth it. Thoughts?

File: js/Heap.js [hunk 1]
```diff
@@ -23,33 +25,40 @@ const track = (event, payload) => {
   } catch (e) {
     console.log('Error calling Heap.track\n', e);
   }
+});
+
+const triggerCrash = () => {
```
Comment (GrandPoohBear) [inline js/Heap.js]: Is there a reason we need this in the actual library code vs. the test code?  I'm assuming this is a _test only_ function...

Also, I think it'd probably be funny if I was looking in the source of a new lib and the first thing I saw was a `triggerCrash` function!!!

Comment (timothyandrew) [reply js/Heap.js]: Good point! I wasn't able to find a way to add the `triggerCrash` function to the `Heap` API from the detox level, so I simply moved to unit tests with Jest instead. These probably cover more of the actual code involved anyway, so I'd say this was for the best.

**Pull request merged**